File: docs-examples/example-java/src/test/java/io/micronaut/rabbitmq/docs/consumer/types/ProductClient.java
package io.micronaut.rabbitmq.docs.consumer.types;

import io.micronaut.context.annotation.Requires;
// tag::imports[]
import io.micronaut.rabbitmq.annotation.Binding;
import io.micronaut.rabbitmq.annotation.RabbitClient;
// end::imports[]

@Requires(property = "spec.name", value = "TypeBindingSpec")
// tag::clazz[]
@RabbitClient // <1>
public interface ProductClient {

    @Binding("product") // <2>
    void send(byte[] data, String contentType); // <3>
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/configuration/rabbitmq/docs/consumer/types/ProductClient.kt
package io.micronaut.configuration.rabbitmq.docs.consumer.types

// tag::imports[]
import io.micronaut.configuration.rabbitmq.annotation.Binding
import io.micronaut.configuration.rabbitmq.annotation.RabbitClient
import io.micronaut.context.annotation.Requires
// end::imports[]

@Requires(property = "spec.name", value = "TypeBindingSpec")
// tag::clazz[]
@RabbitClient // <1>
interface ProductClient {

    @Binding("product") // <2>
    fun send(data: ByteArray, contentType: String)  // <3>
}
// end::clazz[]